app/lib/common/providers/room_providers.dart

```diff
@@ -76,6 +78,62 @@ final briefRoomItemWithMembershipProvider =
   );
 });
 
+final briefRoomItemsWithMembershipProvider =
+    FutureProvider.autoDispose<List<RoomItem>>((ref) async {
+  final chatList =
+      ref.watch(chatsProvider).where((element) => (!element.isDm())).toList();
+
+  List<RoomItem> items = [];
+  for (final element in chatList) {
+    final room =
+        await ref.watch(maybeRoomProvider(element.getRoomIdStr()).future);
+    if (room != null) {
+      final profileData = await ref
+          .watch(roomProfileDataProvider(element.getRoomIdStr()).future);
+
+      final item = RoomItem(
+        roomId: room.roomIdStr(),
+        room: room,
+        membership: room.isJoined() ? await room.getMyMembership() : null,
+        activeMembers: [],
+        roomProfileData: profileData,
+      );
+      items.add(item);
+    }
+  }
+  return items;
+});
+
+final roomSearchedChatsProvider =
+    FutureProvider.autoDispose<List<RoomItem>>((ref) async {
+  final allRooms =
+      ref.watch(chatsProvider).where((element) => (!element.isDm())).toList();
+  final searchValue = ref.watch(chatSearchValueProvider);
+  final foundRooms = List<RoomItem>.empty(growable: true);
+
+  for (final element in allRooms) {
+    final room =
+        await ref.watch(maybeRoomProvider(element.getRoomIdStr()).future);
+    if (room != null) {
+      final profileData = await ref
+          .watch(roomProfileDataProvider(element.getRoomIdStr()).future);
+      final name = profileData.displayName ?? element.getRoomIdStr();
+
+      if (name.toLowerCase().contains(searchValue!.toLowerCase())) {
```

Comment:
How ```searchValue``` null value is handled right now??? This comparison might break the provider itself. How about doing this?
```suggestion
      if(searchValue != null){
      
      if (name.toLowerCase().contains(searchValue!.toLowerCase())) {
         .....
      }
    }
    else {
        throw error
    }
```

Reply:
My Bad!
Updated code which will cover above use-case. :)